### PR TITLE
rustfmt librand/distributions

### PR DESCRIPTION
**Pull request by mseri**
r? @nrc

I was wondering, what is the standard for pattern matching in presence of guards? I liked more the original alignment of `distributions/gamma.rs` line 91 than the new one.

### CODE REVIEW

**Comment by nrc** [comment]
@mseri in general, rustfmt does not try to align tokens between sequential items, for example the `=>` in the match arms here. Sometimes it is desirable, but it is really hard for rustfmt to tell the difference between intentional alignment and whitespace errors. For now, the policy is to accept this and where something looks much better with alignment, do it manually and add skip attributes. I think in this case rustfmt's version is OK.

**Comment by nrc** [comment]
@bors: r+ rollup

Thanks for the PR, this looks great!

**Comment by mseri** [comment]
Ok. Thanks for the heads up. Should I submit a PR to the style guide adding some of the missing style guidelines that I saw while doing these rustfmt (and the ones in preparation)?

**Comment by nrc** [comment]
You can if you like, the style guide is lacking a lot of the detail of rustfmt. I expect we'll revamp it properly at some point in the future

File: src/librand/distributions/mod.rs [hunk 1]
```diff
@@ -103,7 +107,8 @@ impl<'a, T: Clone> WeightedChoice<'a, T> {
     /// - the total weight is larger than a `usize` can contain.
     pub fn new(items: &'a mut [Weighted<T>]) -> WeightedChoice<'a, T> {
         // strictly speaking, this is subsumed by the total weight == 0 case
-        assert!(!items.is_empty(), "WeightedChoice::new called with no items");
+        assert!(!items.is_empty(),
```
Comment (nagisa) [inline src/librand/distributions/mod.rs]: This getting split into two lines makes little sense @nrc.

File: src/librand/distributions/mod.rs [hunk 2]
```diff
@@ -113,25 +118,28 @@ impl<'a, T: Clone> WeightedChoice<'a, T> {
...
-                None => panic!("WeightedChoice::new called with a total weight \
-                               larger than a usize can contain")
+                None => panic!("WeightedChoice::new called with a total weight larger than a \
+                                usize can contain"),
             };
 
             item.weight = running_total;
         }
-        assert!(running_total != 0, "WeightedChoice::new called with a total weight of 0");
+        assert!(running_total != 0,
```
Comment (nagisa) [inline src/librand/distributions/mod.rs]: Ditto, especially that line above was *extended* to 100 characters.

File: src/librand/distributions/mod.rs [hunk 3]
```diff
@@ -298,49 +319,56 @@ mod tests {
...
+                Weighted { weight: 1, item: 54 },
+                Weighted { weight: 1, item: 55 },
+                Weighted { weight: 1, item: 56 }),
            [50, 51, 52, 53, 54, 55, 56]);
     }
 
-    #[test] #[should_panic]
+    #[test]
+    #[should_panic]
+    #[rustfmt_skip]
```
Comment (nagisa) [inline src/librand/distributions/mod.rs]: Is this necessary here?

**Comment by mseri** [comment]
@steveklabnik there is a typo that I fixed this morning. My apologies for that, I ran the tests on the wrong patch branch.

File: src/librand/distributions/mod.rs (hunk 2 above)
Comment (mseri) [inline src/librand/distributions/mod.rs]: I don't know, but I like much less the extension of line 121-122 than splitting the assert.

File: src/librand/distributions/mod.rs (hunk 1 above)
Comment (nrc) [inline src/librand/distributions/mod.rs]: We split when the internals of a function call or function-like macro is more than a heuristic limit (I think 60 chars). Mostly this works well. Occasionally (like here) it seems a bit over the top. I expect its a heuristic we'll polish as time goes by.

File: src/librand/distributions/mod.rs (hunk 2 above)
Comment (nrc) [inline src/librand/distributions/mod.rs]: see above. The string is lengthened out to the max possible. Again, it's not a perfect heuristic, but it should improve (we could try to balance line lengths in string literals, for example).

**Comment by mseri** [comment]
I think we should just get rid of those rustfmt_skip, is there any other way we can tell rustfmt to skip the file?

**Comment by nrc** [comment]
You need to add #![feature(custom_attribute)] and #![allow(unused_attributes)] to the crate root to pass tests.

The skip attribute is the only way to get rustfmt to skip a file or item.

**Comment by mseri** [comment]
Sorry, I messed up with this branch. I am closing the pull request - cleaning everything and resubmitting a clean one.

**Pull request closed without merging**